core/src/main/java/io/kestra/core/runners/Worker.java

```diff
@@ -86,6 +86,8 @@ public class Worker implements Runnable, Closeable {
     @Getter
     private final String workerGroup;
 
+    protected final List<Runnable> subscriptionCancels = new ArrayList<>();
```

Comment:
I don't really understand why we don't have this mechanism already in place tbh. I think that's wrong to say that the Worker right now is closing himself, looks more like he is shutting down the whole app so imo we should have this and cancel our subscription instead of shutting down the topic, WDYT ?

Reply:
Here I did this for the tests since some tests need to be retried but their assert is in the receive method. So I have to stop this continuous assertion for the next test to work, but yeah I'm not a big fan either of adding test-specific code in production code.
Still I think this should be a productionwise thing

Reply:
I didn't want to break anything here that's why I didn't add the cleanup in the worker itself but it should definitely be done I think